### DISCUSSION TITLE
User Access Control Structure?

### DISCUSSION THREAD

**Discussion by jlpoolen** [scottlamb/moonfire-nvr#236] [Ideas]
I know this is way beyond the design criteria, but I am pleased to say that I have been opening my instance of moonfire-nvr to neighbors (using pfSense and NAT routing port 8080 to my raspberry pi) and nearby property owners so they may access the feeds, live and historic, directly.  I create an account for each user.  So the use of the server has become more than a single user/owner.  One neighbor who monitors further west the portion of the alley my cameras cover constantly logs in when something suspicious comes by his cameras (Swann brand) and he wants to get a better shot.  I run my cameras at maximum resolution and frames per second to get the best possible image.  I'm constantly being asked about my system because the quality of the Reolinks at full resolution gives a remarkable picture.  

It would be great if there could be some sort of access control list for each user that might specify what cameras/feeds they may access.  I currently have 11 cameras and will be adding two more, one to cover an adjacent property owner's lot bringing the count to two of cameras devoted to other premises.

I appreciate how complicated access control lists, aka ACLs, can be and can sometimes be counter-productive.

The camera set-up I have coupled with Moonfire-nvr has helped tremendously to discourage problem people - transients using the alley as a bathroom, illegal dumping (small scale, but dumping nonetheless) and drug activity.  We had one problem woman who appears to be stealing gardening stuff, plants, a rose arbor, flowers, and she would ride by between midnight and 6:00 a.m. with her booty, I published a series of her passing by and called her The Midnight Gardener and shared the select videos with neighbors.  Finally a neighbor contacted her landlord as her activity was escalating and the landlord confronted her after he demanded proof and then viewed the videos.  Needles to say, she rarely uses the alley and coincidentally there appears to be a significant reduction of what looks like drug transactions.  I think word got out on the street that the cameras do, indeed, capture everything.  Everyone agrees the alley has improved thanks to my and a couple of neighbors being vigilant with our surveillance.  It is the ability to share the Moonfire-nvr interface via a browser that makes this project sparkle.

I'm starting to sound like a constant cheerleader, but when there is something to cheer about, the cheerleaders cheer.

**Comment by scottlamb** [comment]
I'm really glad it's so useful for you and wishing I've had more time lately to work on this kind of feature.

I made some progress today, though: an admin_user permission which can be used to change users via POST to /users/:id. Currently primarily their password. It won't be hard to extend that to control their permissions. I think it shouldn't be too hard to extend the permission setup to add a view_camera.

**Reply by jlpoolen** [reply]
Worry not.  Moonfire-nvr has allowed me to retrieve footage of a man yanking out a water hydrant connected to the water main (water main was turned off at meter because someone previously had tampered with the hydrant soas to cause a break in the line below).  My Reolink triggers would not have captured the event unless they were capturing everything.  So again, having Moonfire's 24/7 archiving has been invaluable.  See https://salemdata.us/Adventures/RedRidingHoodie/dec_19_22/RedRidingHoodie_Yanking.mp4   

Unfortunately, Salem Police do not do much about property crimes, there's more vicious stuff going on here that has a higher priority.
